src/components/tx/SignOrExecuteForm/hooks.ts

```diff
@@ -164,7 +164,7 @@ export const useRecommendedNonce = (): number | undefined => {
       return getRecommendedNonce(safe.chainId, safeAddress)
     },
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    [safeAddress, safe.chainId, safe.txQueuedTag], // update when tx queue changes
+    [safeAddress, safe.chainId, safe.txQueuedTag, safe.nonce], // update when tx queue or safe nonce changes
```

Comment:
This can be slightly optimized if we just return `Math.max(safe.nonce, recommendedNonce)` from the outer hook, no need to refetch the recommended nonce.